diff --git a/etc/nsjail/compilers-and-tools.cfg b/etc/nsjail/compilers-and-tools.cfg
index a0d5e76..6381be4 100644
--- a/etc/nsjail/compilers-and-tools.cfg
+++ b/etc/nsjail/compilers-and-tools.cfg
@@ -232,16 +232,6 @@ mount {
     is_bind: true
 }
 
-mount {
-    src: "/cefs"
-    dst: "/cefs"
-    is_bind: true
-    # needed for cefs to see new mounts.
-    # needs ce's patched nsjail to work
-    # http://github.com/compiler-explorer/nsjail
-    needs_mount_propagation: true
-}
-
 # Needed for icc
 mount {
     src: "/opt/intel"
diff --git a/etc/nsjail/user-execution.cfg b/etc/nsjail/user-execution.cfg
index 7f13673..0d57f72 100644
--- a/etc/nsjail/user-execution.cfg
+++ b/etc/nsjail/user-execution.cfg
@@ -209,16 +209,6 @@ mount {
     is_bind: true
 }
 
-mount {
-    src: "/cefs"
-    dst: "/cefs"
-    is_bind: true
-    # needed for cefs to see new mounts.
-    # needs ce's patched nsjail to work
-    # http://github.com/compiler-explorer/nsjail
-    needs_mount_propagation: true
-}
-
 # Remove some sensitive stuff from execution
 mount {
     dst: "/opt/compiler-explorer/sonar"
